tools/document-pii/src/main.rs

```diff
@@ -0,0 +1,574 @@
+    }
+
+    if !Path::new(&format!("{}/Cargo.toml", &chosen_dir)).exists() {
+        panic!("Cargo.toml not found. Either run this script from a rust crate/workspace, or pass in a valid path with the -p flag");
+    }
+
+    let paths = find_rs_files(chosen_dir);
+
+    if paths.is_empty() {
+        panic!("No rust files found in {current_dir}");
+    }
+
+    for path in &paths {
+        // first iterate through all rust files to create the hashmap
+        let mut hashmap_filler = FillHashMap {
+            module_path: rust_file_to_use_path(path.as_path()),
+        };
+        let file_content = fs::read_to_string(path.as_path()).unwrap();
+        let syntax_tree: syn::File = syn::parse_file(&file_content).unwrap();
+        hashmap_filler.visit_file(&syntax_tree);
+    }
+
+    match matches.value_of("item") {
+        Some(path) => {
+            if let Some(structorenum) = {
+                let guard = ALL_TYPES.lock().unwrap();
+                guard.get(path).cloned()
+            } {
+                let (theitem, module_path) = structorenum;
+                let mut visitor = TypeVisitor {
+                    module_path,
+                    ..Default::default()
+                };
+                match theitem {
+                    EnumOrStruct::ItemStruct(itemstruct) => visitor.visit_item_struct(&itemstruct),
+                    EnumOrStruct::ItemEnum(itemenum) => visitor.visit_item_enum(&itemenum),
+                };
+            } else {
+                panic!("Please provide a fully qualified path to a struct or an enum. E.g. 'relay_general::protocol::Event'");
+            }
+        }
+        None => {
+            for path in paths {
+                // Then start the processing, with the full hashmap we can recursively go into items
+
+                let file_content = fs::read_to_string(path.as_path()).unwrap();
+                let syntax_tree: syn::File = syn::parse_file(&file_content).unwrap();
+                let module_path = rust_file_to_use_path(path.as_path());
+
+                let mut visitor = TypeVisitor {
+                    module_path,
+                    ..Default::default()
+                };
+
+                visitor.visit_file(&syntax_tree);
+            }
+        }
+    }
+
+    let mut output_vec = vec![];
+    for pii in PII_TYPES.lock().unwrap().iter() {
+        let mut output = String::new();
+        output.push_str(&pii[0].type_name);
+
+        for path in pii {
+            output.push_str(&format!(".{}", path.field_name));
+        }
+
+        let unnamed_replace = match matches.value_of("unnamed").unwrap() {
+            "" => "".to_string(),
+            other => format!("{other}."),
+        };
+
+        output = output.replace("{{Unnamed}}.", &unnamed_replace);
+        output_vec.push(output);
+    }
+    output_vec.sort();
+    for x in &output_vec {
+        println!("{x}");
+    }
+}
+
+fn rust_file_to_use_path(file_path: &Path) -> String {
```

Comment:
IIUC this method attempts to get the `use` path from a rust file, and we are trying to do that by dealing with the file system. Is there a way to get this by information in the AST?

Reply:
unfortunately not, that would have made this task a lot easier